drivers/test.go

```diff
@@ -446,10 +446,13 @@ func (td *TestDriver) AssertBalance(addr address.Address, expected abi_spec.Toke
 }
 
 // Checks an actor's balance and callSeqNum.
-func (td *TestDriver) AssertActor(addr address.Address, balance abi_spec.TokenAmount, callSeqNum uint64) {
+func (td *TestDriver) AssertActor(addr address.Address, prevBalance abi_spec.TokenAmount, rct types.MessageReceipt, callSeqNum uint64) {
```

Comment:
This was definitely intending to assert the balance now, not before. This is getting a bit confusing. 

From looking at the call sites, I think I understand the intention here is to move the expected calculation here. It makes sense when there's no value transfer, as in many cases. It's a little opaque. How about adding a "transferred" parameter here too, so the complete calculation can be inside this method. That would clean up a bunch of cases where you've had to hackily adjust the "prev" balance at the caller including, confusingly, for self-transfers where the net should be zero.

 I think it'll then be fairly clear that the `(prior, transferred, receipt)` params then give the expected post balance.

Reply:
Done (using the receipt's exitcode, as you suggested).

Reply:
I'm expecting to see the type signature change to add a `transferred` parameter.